dashboard/app/models/user.rb

```diff
@@ -151,6 +153,23 @@ class User < ActiveRecord::Base
   has_many :districts_users, class_name: 'DistrictsUsers'
   has_many :districts, through: :districts_users
 
+  belongs_to :school_info
+  accepts_nested_attributes_for :school_info, reject_if: :check_school_info
+  validates_presence_of :school_info, unless: :school_info_optional?
+
+  def check_school_info(school_info_attr)
```

Comment:
This is similar to the deduplication method, in that it does have a side effect (it disables validation, as well as returning a boolean) - I'm now remembering why it's done this way, and it's that this reject_if check seems to be the only thing that runs before :school_info gets validated, so if it needs to be altered, it has to happen in here, as well as return a boolean. I'm open to suggestions about how to make this clearer.